### PR TITLE
feat(data-masking): adapt to task framework

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-feature
module-data masking
module-task framework

#### What this PR does / why we need it:
This PR adapt data-masking in database change task to new task framework.
Details:
1. In `TaskController`, I add a new API `/api/v2/task/querySensitiveColumn` used by executor for querying a `List<Set<DBColumn>> tableRelatedDBColumns` whether has sensitive columns and get `List<MaskingAlgorithm> maskingAlgorithms` for masking query result.
2. The sensitive column extracting and data masking are executed in executor.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Self-test passed in PROCESS model

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java [hunk 1]
```diff
@@ -167,34 +156,30 @@ protected void onProgressUpdate(Long taskId, TaskService taskService) {
...
-        jobData.put(JobParametersKeyConstants.META_TASK_PARAMETER_JSON, JsonUtils.toJson(parameters));
-        jobData.put(JobParametersKeyConstants.CONNECTION_CONFIG, JobUtils.toJson(config));
-        jobData.put(JobParametersKeyConstants.FLOW_INSTANCE_ID, FlowTaskUtil.getFlowInstanceId(execution) + "");
-        jobData.put(JobParametersKeyConstants.CURRENT_SCHEMA, FlowTaskUtil.getSchemaName(execution));
-        jobData.put(JobParametersKeyConstants.SESSION_TIME_ZONE, connectProperties.getDefaultTimeZone());
-        jobData.put(JobParametersKeyConstants.TASK_EXECUTION_TIMEOUT_MILLIS, parameters.getTimeoutMillis() + "");
-        if (CollectionUtils.isNotEmpty(parameters.getSqlObjectIds())) {
+        config.setDefaultSchema(FlowTaskUtil.getSchemaName(execution));
+        taskParameters.setConnectionConfig(config);
+        taskParameters.setFlowInstanceId(FlowTaskUtil.getFlowInstanceId(execution));
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java:166]: Okay!

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobUrlConstants.java [hunk 2]
```diff
@@ -26,6 +26,8 @@ public class JobUrlConstants {
 
     public static final String TASK_HEART = "/api/v2/task/heart";
 
+    public static final String TASK_QUERY_SENSITIVE_COLUMN = "/api/v2/task/querySensitiveColumn";
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobUrlConstants.java:29]: It is used!

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java [hunk 3]
```diff
@@ -108,29 +105,21 @@ public boolean isCancelled() {
...
-        isSuccessful = status == JobStatus.DONE;
+        JobEntity jobEntity = taskFrameworkService.find(this.jobId);
+        isSuccessful = jobEntity.getStatus() == JobStatus.DONE;
         isFailure = !isSuccessful;
-        log.info("Async task ends, taskId={}, activityId={}, returnVal={}, timeCost={}, jobId={}, jobStatus={}",
-                taskId, execution.getCurrentActivityId(), result,
-                System.currentTimeMillis() - getStartTimeMilliSeconds(), jobEntity.getId(), jobEntity.getStatus());
-
-        return result;
+        log.info("Database change task ends, taskId={}, activityId={}", taskId, execution.getCurrentActivityId());
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java:121]: Get, I add necessary log print.

**Review by krihy — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**